src/components/issue-blurb.jsx

```diff
@@ -10,7 +10,7 @@ import GithubFlavoredMarkdown from './gfm';
 import Loadable from './loadable';
 import ColoredIcon from './colored-icon';
 
-const IssueOrPullRequestBlurb = React.createClass({
+class IssueOrPullRequestBlurb extends Component {
```

Comment:
More functional component?

Reply:
Using functional component would lead to an error as it is using `this.onClickIcon` here(https://github.com/li-boxuan/gh-board/blob/073b6e282ec08dfdbe382501673165b8179af044/src/components/issue-blurb.jsx#L61). To be honest, I don't know where the function `onClickIcon` comes from. I use grep and find out it's being used in issue-blurb.jsx but defined nowhere.

Reply:
Ah ok. **We'll leave it as it is.**

But does it even matter? Since `this.onClickIcon` is undefined whether it is using class or functional. It would lead to an error either way.

Probably open an issue after finding where the component is located in the UI, if there's an error.

Reply:
I just transformed it into the functional component. You are right, the undefined `onClickIcon` has nothing to do with whether it is using class or function.
Filed another issue https://github.com/coala/gh-board/issues/29 for the undefined `onClickIcon`.